docs/features/incremental-generators.cookbook.md

```diff
@@ -636,7 +636,188 @@ TODO: https://github.com/dotnet/roslyn/issues/72149
+
+    public void Initialize(IncrementalGeneratorInitializationContext context)
+    {
+        context.RegisterPostInitializationOutput(ctx =>
+        {
+            //Generate the AutoImplementProperties Attribute
+            const string autoImplementAttributeDeclarationCode = $$"""
+// <auto-generated/>
+using System;
+namespace {{AttributeNameSpace}};
+
+[AttributeUsage(AttributeTargets.Class, Inherited = false, AllowMultiple = false)]
+sealed class {{AttributeClassName}} : Attribute
+{
+    public Type[] InterfacesTypes { get; }
+    public {{AttributeClassName}}(params Type[] interfacesTypes)
+    {
+        InterfacesTypes = interfacesTypes;
+    }
+}
+""";
+            ctx.AddSource($"{AttributeClassName}.g.cs", autoImplementAttributeDeclarationCode);
+        });
+
+        IncrementalValuesProvider<ClassModel> provider = context.SyntaxProvider.ForAttributeWithMetadataName(
+            fullyQualifiedMetadataName: FullyQualifiedAttributeName,
+            predicate: static (node, cancellationToken_) => node is ClassDeclarationSyntax,
+            transform: static (ctx, cancellationToken) =>
+            {
+                ISymbol classSymbol = ctx.SemanticModel.GetDeclaredSymbol(ctx.TargetNode, cancellationToken)!;
+
+                return new ClassModel(
+                    classSymbol.Name,
+                    classSymbol.ContainingNamespace.ToDisplayString(),
+                    GetInterfaceModels(classSymbol)
+                    );
+            }).Where(m => m is not null);
+
+        context.RegisterSourceOutput(provider, static (context, classModel) =>
+        {
+            foreach (InterfaceModel interfaceModel in classModel.Interfaces)
+            {
+                StringBuilder sourceBuilder = new($$"""
+                    // <auto-generated/>
+                    namespace {{classModel.NameSpace}};
+
+                    public partial class {{classModel.Name}} : {{interfaceModel.FullyQualifiedName}}
+                    {
+
+                    """);
+
+                foreach (string property in interfaceModel.Properties)
+                {
+                    sourceBuilder.AppendLine(property);
+                }
+
+                sourceBuilder.AppendLine("""
+                    }
+                    """);
+
+                //Concat class name and interface name to have unique file name if a class implements two interfaces with AutoImplement Attribute
+                string generatedFileName = $"{classModel.Name}_{interfaceModel.FullyQualifiedName}.g.cs";
+                context.AddSource(generatedFileName, sourceBuilder.ToString());
+            }
+        });
+    }
+
+    private static EquatableList<InterfaceModel> GetInterfaceModels(ISymbol classSymbol)
+    {
+        EquatableList<InterfaceModel> ret = [];
+
+        AttributeData attribute = classSymbol.GetAttributes().First(a => a.AttributeClass!.Name == AttributeClassName);
```

Comment:
Just get this from the context. The `Attributes` property will have all the matched attributes; since `AllowMultiple` is false, you can just access the `[0]`th index and not care about any duplicates with errors.